src/sentry/static/sentry/app/components/listViewContainer.jsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+
+import ListView from 'app/components/listView';
+import ListFilters from 'app/components/listFilters';
+import ClimsTypes from 'app/climsTypes';
+import SentryTypes from 'app/sentryTypes';
+import {browserHistory} from 'react-router';
+
+// A wrapper around the presentational components ListView, ListFilters and Pagination,
+// that handles actions in the view
+//
+// TODO: Use this in the substance view. It's based on that.
+
+class ListViewContainer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onSavedSearchCreate = this.onSavedSearchCreate.bind(this);
+    this.onGroup = this.onGroup.bind(this);
+    this.onSort = this.onSort.bind(this);
+    this.onSearch = this.onSearch.bind(this);
+    this.toggleAll = this.toggleAll.bind(this);
+    this.onCursor = this.onCursor.bind(this);
+
+    let {search, groupBy} = this.props;
+    let {cursor} = this.props.listViewState.pagination;
+
+    const query = this.props.location.query;
+    if (query) {
+      search = query.search ? query.search : search;
+      cursor = query.cursor ? query.cursor : cursor;
+      groupBy = query.groupBy ? query.groupBy : groupBy;
+    }
+
+    this.onSearch(search, groupBy, cursor);
+  }
+
+  onSearch(search, groupBy, cursor) {
+    // Corresponding in substance search:
+    // const isGroupHeader = groupBy !== 'substance';
+    // this.props.substanceSearchEntriesGet(search, groupBy, cursor, isGroupHeader);
+
+    this.props.getEntries(search, groupBy, cursor); // TODO: not implemented for workbatch
```

Comment:
getEntries should be included in propTypes.

Reply:
Right - didn't get a warning about that in my toolchain, I'm going to review this a bit better and ping you when it's fixed.

Reply:
:+1: